### ISSUE TITLE
Slowness still persistent even after regexpengine workaround

### ISSUE THREAD

**Issue by Utagai** [myusuf3/numbers.vim#65]
Hi, I've been experiencing some pretty bad lag with vim when editing files at around 400-500~ lines or greater length. I've found the previous issue (#53), and tried the workaround but it doesn't help much. However, toggling relative line numbers off _completely_ solves the issue. 

I'm also running on a pretty powerful machine here. 3.5 GHz quad core, 16 GB ram, running Ubuntu. When just scrolling down in the file, I see an entire 100% usage of a core on the CPU.

**Comment by myusuf3** [comment]
Are you sure its numbers.vim? Did you profile?

**Comment by Utagai** [comment]
Yup, I have. Exhaustively went through, disabling all my plugins (I don't have many to be honest). The moment I disabled numbers `:NumbersToggle` I saw an immediate speed increase back to normal levels (i.e., no lag).

**Comment by myusuf3** [comment]
Thats a vim issue then, I would look into that.

**Comment by Utagai** [comment]
Sorry for the late response here, but mind elaborating on why its a vim issue? Still have this problem, but I've been avoiding it by toggling line numbers off for larger files.

As far as I can tell, it seems that the root cause is indeed numbers.vim, since the problem goes away when I toggle off numbers, but not when I disable any other plugins.

I'd love to get to the bottom of this, since I love numbers.vim and I hate having to disable it just to work on larger files.

**Comment by N4M3Z** [comment]
Just to confirm, I am having the same problem with large files.

**Comment by myusuf3** [comment]
Closing this with some actual measurements, nine years late - apologies for that.

Benchmark: 400 `j`-motions with a forced redraw over a 10,000-line Python file, syntax on, 5 reps, minimum time.

| | nvim 0.12.4 | vim 9.2 |
|---|---|---|
| no numbers | 17.8 ms | 18.5 ms |
| absolute only | 17.8 ms | 19.0 ms |
| hybrid, set by hand | 19.7 ms | 21.7 ms |
| hybrid, plugin loaded | 20.1 ms | 22.2 ms |

The plugin adds ~0.5 ms per 400 motions over setting `number`+`relativenumber` by hand, which is measurement noise. That fits how it works: there is no `CursorMoved` autocmd, so nothing in this plugin runs while you hold `j`.

What does cost something is `relativenumber` itself - every cursor move redraws the whole gutter. On Vim 7.4 that redraw re-ran syntax highlighting through the old regexp engine, which is why you were already on the `regexpengine` workaround from #53 and why it felt like a stall rather than a few percent.

Caveat: I measured on current versions, not on 7.4, so I cannot reproduce your original setup. If this still bites on a current Vim or Neovim, please reopen with the file and I'll dig in again.

If you do hit it on a huge file, `:NumbersDisable` is now a clean escape hatch - as of #78 it leaves plain absolute numbers rather than stripping the gutter.

cc @N4M3Z
